src/rules/no-duplicate-keys.js

```diff
@@ -36,7 +48,7 @@ export default {
 						? node.name.value
 						: node.name.name;
 
-				if (keys.has(key)) {
+				if (keys?.has(key)) {
```

Comment:
[Question] Is `keys` actually nullish here? Is it possible to hit a `Member` node without having hit an earlier `Object` node?

If not, I'd personally use a type assertion to avoid adding an unnecessary runtime optional chain:

```suggestion
				if (/** @type {Map<string, MemberNode>} */ (keys).has(key)) {
```

Aside: I wish we had a direct equivalent to `!`, but: https://github.com/microsoft/TypeScript/issues/23405.